### PR TITLE
Add visualbert onnx suport

### LINKED ISSUE #555: Community contribution -  `optimum.exporters.onnx` support for new models!
Following what was done by @ChainYo in Transformers, in the [ONNXConfig: Add a configuration for all available models](https://github.com/huggingface/transformers/issues/16308) issue, the idea is to add support for exporting new models in `optimum.exporters.onnx`.

This issue is about the working group specially created for this task. If you are interested in helping out, reply here, take a look at this [organization](https://huggingface.co/OWG), or add  `ChainYo#3610` on discord.

We want to contribute to Hugging Face's ONNX export implementation for all available models on Hugging Face Hub. There are already a lot of architectures implemented for converting PyTorch models to ONNX, but we need more! We need them all!

Feel free to join us in this adventure! Join the org by clicking [here](https://huggingface.co/organizations/OWG/share/TskjfGaGjGnMXXssbPPXrQWEIbosGqZshZ)

Here is a non-exhaustive list of models that all models available:

- [x] Albert
- [x] BART
- [x] BeiT
- [x] BERT
- [ ] BigBird (Critical issue: https://github.com/huggingface/optimum/issues/754#issuecomment-1429830467)
- [ ] BigBirdPegasus (Critical issue: https://github.com/huggingface/optimum/issues/754#issuecomment-1429830467)
- [x] Blenderbot
- [x] BlenderbotSmall
- [ ] BLIP-2
- [x] BLOOM
- [x] CamemBERT
- [ ] CANINE
- [x] CLIP
- [x] CodeGen
- [x] ConvNext
- [x] ConvBert
- [ ] CTRL
- [x] CvT
- [x] Data2VecText
- [x] Data2VecVision
- [x] Deberta
- [x] DebertaV2
- [x] DeiT
- [ ] DecisionTransformer
- [x] DETR
- [x] Distilbert
- [ ] DPR
- [x] DPT
- [x] ELECTRA
- [ ] FNet
- [ ] FSMT
- [x] Flaubert
- [ ] FLAVA
- [ ] Funnel Transformer
- [x] GLPN
- [x] GPT2
- [x] GPTJ
- [x] GPT-Neo
- [x] GPT-NeoX
- [x] Hubert
- [x] I-Bert
- [x] ImageGPT 🛠️ @adit299 
- [ ] LED
- [x] LayoutLM
- [ ] LayoutLMv2 (but 🛠️ in Transformers)
- [x] LayoutLMv3
- [ ] LayoutXLM
- [ ] LED
- [x] LeViT
- [ ] 🛠️ Longformer (Critical issue: https://github.com/huggingface/optimum/issues/776#issuecomment-1429680121)
- [x] LongT5
- [ ] Luke (but 🛠️ in Transformers)
- [ ] Lxmert
- [x] M2M100
- [ ] MaskFormer
- [x] mBart
- [ ] MCTCT
- [x] MPNet
- [x] MT5
- [x] MarianMT
- [ ] MegatronBert
- [x] MobileBert
- [x] MobileViT
- [ ] Nyströmformer
- [x] OpenAIGPT-2
- [x] OPT ((but 🛠️ in Transformers)
- [x] OWLViT
- [x] Pix2Struct
- [x] PLBart
- [x] Pegasus
- [x] Perceiver
- [x] PoolFormer
- [ ] ProphetNet
- [ ] QDQBERT
- [ ] RAG
- [ ] REALM
- [ ] Reformer (but 🛠️ in Transformers)
- [x] RemBert
- [x] ResNet
- [x] RegNet 🛠️ @asrimanth 
- [ ] RetriBert
- [x] RoFormer
- [x] RoBERTa
- [x] SEW
- [x] SEW-D
- [x] SegFormer
- [x] Speech2Text
- [ ] Speech2Text2
- [x] Splinter
- [x] SqueezeBERT
- [x] Swin Transformer
- [x] T5
- [ ] TAPAS 🛠️ @someshfengde 
- [ ] TAPEX
- [ ] Transformer XL
- [x] TrOCR
- [ ] UniSpeech
- [ ] UniSpeech-SAT
- [ ] VAN
- [x] ViT
- [ ] Vilt
- [ ] VisualBERT
- [x] Wav2Vec2
- [x] WavLM
- [x] Whisper
- [ ] XGLM
- [x] XLM
- [ ] XLMProphetNet
- [x] XLM-RoBERTa
- [x] XLM-RoBERTa-XL
- [ ]  XLNet (but 🛠️ in Transformers)
- [x] YOLOS
- [ ] Yoso

🛠️ next to a model suggests that the PR is in progress. If there is nothing next to a model, it means that ONNX does not yet support the model, and thus we need to add support for it.

If you need help implementing an unsupported model, here is a [guide](https://huggingface.co/docs/optimum/main/en/exporters/onnx/usage_guides/contribute) from HuggingFace Optimum documentation.

### PR DESCRIPTION
**Pull request by tedasdf**
# What does this PR do?

Fixes #555 (issue)

Support VisualBERT model config.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by echarlaix — COMMENTED** [review]
Thanks for your contribution @tedasdf !

File: optimum/exporters/tasks.py [hunk 1]
```diff
@@ -1108,6 +1108,12 @@ class TasksManager:
             "text-to-audio",
             onnx="VitsOnnxConfig",
         ),
+        "visualbert": supported_tasks_mapping(
```
[the remark this answers is not in the document]
Comment (tedasdf) [reply optimum/exporters/tasks.py:1111]: wouldnt that be visual_bert

File: tests/exporters/exporters_utils.py [hunk 2]
```diff
@@ -286,6 +287,7 @@
     "speech-to-text": "codenamewei/speech-to-text",
     "xlm": "xlm-clm-ende-1024",
     "xlm-roberta": "Unbabel/xlm-roberta-comet-small",
+    "visualbert": "unc-nlp/visualbert-uncased",
```
[the remark this answers is not in the document]
Comment (tedasdf) [reply tests/exporters/exporters_utils.py:290]: should i also change visualbert into visual_bert in reference to the config.json

**Comment by tedasdf** [comment]
Hi @echarlaix, I wanted to thank you for reviewing the pull request. Frankly, I’m struggling a bit. I understand the files that need to be altered and the function to be implemented, but I’m having trouble finding the information related to the model, especially given that I’m not very familiar with such a large codebase. I would like to know if there are ways I can contribute more to this issue. I understand you’re busy, but if possible, could you guide me further?

I found the structure of inputs and outputs from the forward function of the model [here](https://huggingface.co/docs/transformers/en/model_doc/visual_bert). However, I’m not sure if the model needs the normalized text configuration or where to locate the appropriate _task_mapping attribute. I chose these areas based on the tasks the model has been fine-tuned for, but I could use some guidance on where to look.

File: tests/exporters/exporters_utils.py [hunk 3]
```diff
@@ -197,6 +197,7 @@
             "document-question-answering-with-past",
         ],
     },
+    "visual_bert": "hf-internal-testing/tiny-random-VisualBertModel",
```
Comment (echarlaix) [inline tests/exporters/exporters_utils.py:200]:
```suggestion
    "visual-bert": "hf-internal-testing/tiny-random-VisualBertModel",
```

File: tests/exporters/exporters_utils.py [hunk 4]
```diff
@@ -286,6 +287,7 @@
     "speech-to-text": "codenamewei/speech-to-text",
     "xlm": "xlm-clm-ende-1024",
     "xlm-roberta": "Unbabel/xlm-roberta-comet-small",
+    "visual_bert": "uclanlp/visualbert-vqa-coco-pre",
```
Comment (echarlaix) [inline tests/exporters/exporters_utils.py:290]:
```suggestion
    "visual-bert": "uclanlp/visualbert-vqa-coco-pre",
```

File: optimum/exporters/tasks.py [hunk 5]
```diff
@@ -1108,6 +1108,12 @@ class TasksManager:
             "text-to-audio",
             onnx="VitsOnnxConfig",
         ),
+        "visual_bert": supported_tasks_mapping(
```
Comment (echarlaix) [inline optimum/exporters/tasks.py:1111]:
```suggestion
        "visual-bert": supported_tasks_mapping(
```

File: optimum/exporters/tasks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (echarlaix) [reply optimum/exporters/tasks.py]: model_type is defined as follows https://github.com/huggingface/optimum/blob/01110adf076c94e395d1472a760eafac2c0a73aa/optimum/exporters/tasks.py#L1318 so here `"visual-bert"`

**Comment by echarlaix** [comment]
Hi @tedasdf, thanks for your contribution! I see that you closed this PR yesterday but happy to help if you want to re-open it and continue working on the integration. 

> I found the structure of inputs and outputs from the forward function of the model [here](https://huggingface.co/docs/transformers/en/model_doc/visual_bert). However, I’m not sure if the model needs the normalized text configuration or where to locate the appropriate _task_mapping attribute. I chose these areas based on the tasks the model has been fine-tuned for, but I could use some guidance on where to look.

The inputs used to export the model needs to be a subset of the model's inputs (a check is done here https://github.com/huggingface/optimum/blob/v1.23.3/optimum/exporters/onnx/convert.py#L96 for this). So you need to first make sure that the inputs defined in your `VisualBertOnnxConfig` is a subset of https://github.com/huggingface/transformers/blob/v4.46.3/src/transformers/models/visual_bert/modeling_visual_bert.py#L700-L712

In your case I see that you have `pixel_values` which is not part of the forward signature, do you mean `visual_embeds` ? If yes, you'll need to rename it using the `rename_ambiguous_inputs` method. Also `pixel_values` inputs are generated as tensors of shape [`batch_size`, `num_channels`, `height`, `width`]
https://github.com/huggingface/optimum/blob/d6de6762e0e4bf8136f0435211a0e777f5bf2f33/optimum/utils/input_generators.py#L829 which I don't think is what you expect from `visual_embeds`.
